tests/windows_test.go

```diff
@@ -0,0 +1,241 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package tests_test
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/tests"
+)
+
+const (
+	windowsDisk       = "windows-disk"
+	windowsFirmware   = "5d307ca9-b3ef-428c-8861-06e72d69f223"
+	windowsVm         = "windows-vm"
+	windowsVmUser     = "Administrator"
+	windowsVmPassword = "Heslo123"
+)
+
+const (
+	winrmCli    = "winrmcli"
+	winrmCliCmd = "/go/bin/winrm-cli"
+)
+
+var _ = Describe("Windows VM", func() {
+	flag.Parse()
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	tests.PanicOnError(err)
+
+	gracePeriod := int64(0)
+	spinlocks := uint32(8191)
+	firmware := types.UID(windowsFirmware)
+	_false := false
+
+	windowsVm := &v1.VirtualMachine{
+		ObjectMeta: metav1.ObjectMeta{Name: windowsVm},
+		Spec: v1.VirtualMachineSpec{
+			TerminationGracePeriodSeconds: &gracePeriod,
+			Domain: v1.DomainSpec{
+				CPU: &v1.CPU{Cores: 2},
+				Features: &v1.Features{
+					ACPI: v1.FeatureState{},
+					APIC: &v1.FeatureAPIC{},
+					Hyperv: &v1.FeatureHyperv{
+						Relaxed:   &v1.FeatureState{},
+						VAPIC:     &v1.FeatureState{},
+						Spinlocks: &v1.FeatureSpinlocks{Retries: &spinlocks},
+					},
+				},
+				Clock: &v1.Clock{
+					ClockOffset: v1.ClockOffset{UTC: &v1.ClockOffsetUTC{}},
+					Timer: &v1.Timer{
+						HPET:   &v1.HPETTimer{Enabled: &_false},
+						PIT:    &v1.PITTimer{TickPolicy: v1.PITTickPolicyDelay},
+						RTC:    &v1.RTCTimer{TickPolicy: v1.RTCTickPolicyCatchup},
+						Hyperv: &v1.HypervTimer{},
+					},
+				},
+				Firmware: &v1.Firmware{UUID: firmware},
+				Resources: v1.ResourceRequirements{
+					Requests: k8sv1.ResourceList{
+						k8sv1.ResourceMemory: resource.MustParse("4096Mi"),
+					},
+				},
+				Devices: v1.Devices{
+					Disks: []v1.Disk{
+						{
+							Name:       windowsDisk,
+							VolumeName: windowsDisk,
+							DiskDevice: v1.DiskDevice{Disk: &v1.DiskTarget{Bus: "sata"}},
+						},
+					},
+				},
+			},
+			Volumes: []v1.Volume{
+				{
+					Name: windowsDisk,
+					VolumeSource: v1.VolumeSource{
+						Ephemeral: &v1.EphemeralVolumeSource{
+							PersistentVolumeClaim: &k8sv1.PersistentVolumeClaimVolumeSource{
+								ClaimName: tests.DiskWindows,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	tests.BeforeAll(func() {
+		windowsPv, err := virtClient.CoreV1().PersistentVolumes().Get(tests.DiskWindows, metav1.GetOptions{})
+		if err != nil {
+			Skip(fmt.Sprintf("Skip Windows tests that requires PVC %s", tests.DiskWindows))
+		}
+		windowsPv.Spec.ClaimRef = nil
```

Comment:
What does this do?

Reply:
I assume that before testing we already added windows PV to the cluster, and maybe it already used by some PVC(currently not), so the PV will have the state `Released`, but new PVC still cannot use it(https://kubernetes.io/docs/concepts/storage/persistent-volumes/#retain), so I remove `ClaimRef` to move it to `Available` state(I checked via google, looks it acceptable action that does not do any harm to the cluster).
I think I will check on PV state, if it has `Released` state I will remove `ClaimRef`, otherwise will skip windows tests.